### DISCUSSION TITLE
Misformed armored text openpgpg problem

### DISCUSSION THREAD

**Discussion by Anonymouseye** [openpgpjs/openpgpjs#1554] [Q&A]
Openpgp always crash after it prints a lot of text it says " Misformed armored".I am new in this library.Can anyone tell me what does that mean?

Code:

> 
> const openpgp = require("openpgp");
> (async () => {
>     const publicKeyArmored = `-----BEGIN PGP PUBLIC KEY BLOCK-----
> 
>     mykey
>     -----END PGP PUBLIC KEY BLOCK-----`; // Public key
> 
>     const privateKeyArmored = `-----BEGIN PGP PRIVATE KEY BLOCK-----
> 
>    mykey
>     -----END PGP PRIVATE KEY BLOCK-----
>     `; // Encrypted private key
>     const passphrase = `mypass`; // Password that private key is encrypted with
> 
>     const publicKey = await openpgp.readKey({ armoredKey: publicKeyArmored });
> 
>   
> 
> })();

**Comment by larabr** [comment]
Make sure that there are not tabs or other whitespaces on the left of the armored lines. For instance this won't work:
```js
const key = `-----BEGIN PGP PUBLIC KEY BLOCK-----

    xjMEYvTupBYJKwYBBAHaRw8BAQdAv6f7gguk8LH71eGqTXDlmS63+lkl6vPT
    vVsZy7bHnFzNDzxhQHdvcmtlci50ZXN0PsKMBBAWCgAdBQJi9O6kBAsJBwgD
    FQgKBBYAAgECGQECGwMCHgEAIQkQTIonUWJ5WWQWIQSyxc91w343HjB5FTdM
    iidRYnlZZHLUAPsHn/LMA9XuVdz33TUCe7kDSakxePFQwDZ0rj3evRXGsQD/
    VeF1Hy6iGgGBJl/IQI/0biaI46QJ7x2joUwQApMloQ3OOARi9O6kEgorBgEE
    AZdVAQUBAQdAln1rHGbf5/KjPjW8x4hp9V3nYTVVt37IEoRtID6HGwUDAQgH
    wngEGBYIAAkFAmL07qQCGwwAIQkQTIonUWJ5WWQWIQSyxc91w343HjB5FTdM
    iidRYnlZZL/GAP49r5AzvFgml5SLSiMnajn1xd5xkQSP9uUgMMCHsje/UwD/
    WIvYE+ikk3TofK6c7VxwyHQddlJgJDW8vrXckoytQw8=
    =2ND4
    -----END PGP PUBLIC KEY BLOCK-----`
```
